### ISSUE TITLE
Custom eslint rule causes error when running linter

### ISSUE THREAD

**Issue by phmz** [nrwl/nx#17006]
### Current Behavior

When I run the commands `nx run api:lint` or `nx affected:lint`, I receive the following error:
>  NX   Failed to load plugin '@nx' declared in 'apps/api/.eslintrc.json » ../../.eslintrc.json#overrides[1]': fn is not a function
I've created a repository with the issue: https://github.com/phmz/nx-linter

### Expected Behavior

I expect the custom eslint rule to work without any errors when running the "nx run api:lint" or "nx affected:lint" commands.

### GitHub Repo

https://github.com/phmz/nx-linter

### Steps to Reproduce

1. Clone the repository: https://github.com/phmz/nx-linter
2. Run `pnpm install` to install dependencies
3. Run `nx run api:lint`

or

1. Clone the repository: https://github.com/nrwl/nx-examples
2. Run `npm install` to install dependencies
3. `nx g @nx/linter:workspace-rule custom-rule`  
4. Run `nx run cart:lint`

### Nx Report

```shell
Node   : 16.14.2
   OS     : darwin arm64
   pnpm   : 8.4.0
   Hasher : Native

   nx (global)        : 16.1.1
   nx                 : 16.1.4
   @nx/js             : 16.1.4
   @nx/jest           : 16.1.4
   @nx/linter         : 16.1.4
   @nx/workspace      : 16.1.4
   @nx/cypress        : 16.1.4
   @nx/eslint-plugin  : 16.1.4
   @nx/nest           : 16.1.4
   @nx/node           : 16.1.4
   @nx/react          : 16.1.4
   @nx/vite           : 16.1.4
   @nx/webpack        : 16.1.4
   typescript         : 5.0.4
```

### Failure Logs

```shell
> nx run api:lint


Linting "api"...

 >  NX   Failed to load plugin '@nx' declared in 'apps/api/.eslintrc.json » ../../.eslintrc.json#overrides[1]': fn is not a function

   Referenced from: /Users/dev/nx-linter/.eslintrc.json

TypeError: Failed to load plugin '@nx' declared in 'apps/api/.eslintrc.json » ../../.eslintrc.json#overrides[1]': fn is not a function
Referenced from: /Users/dev/nx-linter/.eslintrc.json
    at /Users/dev/nx-linter/node_modules/.pnpm/packages/nx/src/plugins/js/utils/register.ts:33:7
    at /Users/dev/nx-linter/node_modules/.pnpm/@nx+eslint-plugin@16.1.4_@swc-node+register@1.4.2_@swc+core@1.3.51_@typescript-eslint+parser@_awkg7lanhpktks2dbrb5wbeojm/packages/eslint-plugin/src/resolve-workspace-rules.ts:32:7
    at Object.<anonymous> (/Users/dev/nx-linter/node_modules/.pnpm/@nx+eslint-plugin@16.1.4_@swc-node+register@1.4.2_@swc+core@1.3.51_@typescript-eslint+parser@_awkg7lanhpktks2dbrb5wbeojm/packages/eslint-plugin/src/resolve-workspace-rules.ts:35:3)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Object.newLoader (/Users/dev/nx-linter/node_modules/.pnpm/pirates@4.0.5/node_modules/pirates/lib/index.js:141:7)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

The error can be fixed by either removing `@swc-node/register` package or using the `1.6.5` version instead of `1.4.2`. The error also happens in node version `v18.16.0`.

**Comment by kfrederix** [comment]
Hello, I had exactly the same issue with Nx 16.1.4 but in my case on a windows machine.
When inspecting the `nx` package in my node_modules, I noticed that the `default.js` file was missing from `@nx/workspace/tasks-runners` (both default.d.ts and default.js.map were there but not default.js)

After **clearing my yarn cache and npm cache** and then removing `node_modules` followed by a fresh yarn install, the issue was resolved and it works fine now.

**Comment by meeroslav** [comment]
This issue has been fixed recently (in version 16.2.0). Please update to latest version.
